### DISCUSSION TITLE
Using `x-mask` with a number input throws error in console

### DISCUSSION THREAD

**Discussion by ayevexy** [alpinejs/alpine#3946] [7. Bugs (archived)]
Using `x-mask` with a number input: `<input x-data x-mask="99" type="number">` works fine, but the following error is thrown on the console when user types:
![image](https://github.com/alpinejs/alpine/assets/124632086/a8186b02-d5fe-4175-bbd5-72016d6b0299)

**Comment by ekwoka** [comment]
That's a strange error, but x-madk is definitely only meant for text inputs.

You can do a mask like you mention easily for normal input limits for number inputs. Max/min/step etc.
